packages/components/src/components/forms/BulkAddUpdateForm.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { List } from 'immutable';
+
+import { QueryInfoForm } from './QueryInfoForm';
+import { capitalizeFirstChar, getCommonDataValues } from '../../util/utils';
+import { QueryColumn, QueryGridModel, QueryInfo } from '../base/models/model';
+import { getEditorModel } from "../../global";
+import { EditorModel } from "../../models";
+
+
+interface Props {
+    selectedRowIndexes: List<number>
+    singularNoun?: string
+    pluralNoun?: string
+    title?: string
+    columnFilter?: (col?: QueryColumn) => boolean
+    itemLabel?: string
+    model: QueryGridModel,
+    onCancel: () => any,
+    onError?: (message: string) => any,
+    onComplete: (data: any, submitForEdit: boolean) => any
+    onSubmitForEdit: (updateData: any) => any
+}
+
+export class BulkAddUpdateForm extends React.Component<Props, any> {
+
+    static defaultProps = {
+        singularNoun: 'row',
+        pluralNoun: 'rows',
+        itemLabel: 'table'
+    };
+
+    getEditorModel(): EditorModel {
+        const { model } = this.props;
+        return getEditorModel(model.getId());
+    }
+
+    getSelectionNoun(): string {
+        const { singularNoun, pluralNoun, selectedRowIndexes } = this.props;
+        return selectedRowIndexes.size === 1 ? singularNoun : pluralNoun;
+    }
+
+    getTitle(): string {
+        const { title, selectedRowIndexes } = this.props;
+        return title ? title : 'Update ' + selectedRowIndexes.size + ' ' + this.getSelectionNoun();
+    }
+
+    getInsertQueryInfo(): QueryInfo {
+        const { model } = this.props;
+        const updateColumns = model.queryInfo.columns.filter((column) => (column.shownInInsertView));
+        return model.queryInfo.set('columns', updateColumns) as QueryInfo;
+    }
+
+    render() {
+        const { pluralNoun, model, onCancel, onComplete, columnFilter, onSubmitForEdit, selectedRowIndexes } = this.props;
+
+        const editorModel = this.getEditorModel();
+        const editorData = editorModel.getRawData(model).filter((val, index) => {
+            return selectedRowIndexes.contains(index);
+        }).toMap();
+        const fieldValues = getCommonDataValues(editorData);
+
+        return (
+            <QueryInfoForm
```

Comment:
I think we're going to want to use the enabling and disabling of fields in the form.  I'm seeing that if I edit samples the parent values don't come through and fields that I don't want to change (that may be different for the different samples I've selected) are cleared.

Reply:
updated